kivy/uix/anchorlayout.py

```diff
@@ -84,28 +84,31 @@ def do_layout(self, *largs):
         anchor_y = self.anchor_y
         padding = self.padding
 
+        left, top, right, bottom = range(4)
```

Comment:
This should be `padd_left, padd_top, padd_right, padd_bottom = self.padding`. Similarly, it should be `shw, shh = self.size_hint` rather than `horizontal, vertical = range(2)` and then indexed.